# Project Summary
This project concerns the well-known Fisher’s Iris data set. You must research the data set and write documentation and code (in Python) to investigate it. An online search for information on the data set will convince you that many people have investigated it previously. You are expected to be able to break this project into several smaller tasks that are easier to solve, and to plug these together after they have been completed.
You might do that for this project as follows:
1. Research the data set online and write a summary about it in your README.
2. Download the data set and add it to your repository.
3. Write a program called analysis.py that:
    1. Outputs a summary of each variable to a single text file,
    2. Saves a histogram of each variable to png files, and
    3. Outputs a scatter plot of each pair of variables.
    4. Performs any other analysis you think is appropriate.

### Importing of Python Standard Libraries
As the first step I will import the standard Python libraries I will be using to aid with my data analysis of the Iris dataset. These libraries will enable me to access standard functions I will need for manipulating the data, making calculations and plotting my data. 

In [1]:
# Import Standard Libraries
# Import Pandas library for working with dataframes
import pandas as pd

#Import machine learning library
import sklearn as skl

# Import Numpy library to work with arrays
import numpy as np

# Import library to visualize data
import matplotlib.pyplot as plt

#Import display function if not already standard
from IPython.display import display

#import random module to make random numbers
import random

#import seaborn library for data visualization & pair plot
import seaborn as sns

### Task 1 -  Research the data set online and write a summary about it in your README.
Please refer to the README.md file located within this repository for the summary on this data set

### Resources used for completion of this section:
1. 

### Task 2 - Download the data set and add it to your repository.
Please refer to the iris.data file located within this repository for the original data file downloaded from the source. 

Read in the information from the iris.data file into this notebook. Note I edited the iris.data file to include the column headings of each variable and at the end the class name as the original file did not have this information. I got the order of each variable (1st to last) from the iris.names file which was also included in the original file download. 

### Resources used for completion of this section:
1. Link to original Iris data set which I downloaded and saved in my repository - https://archive.ics.uci.edu/dataset/53/iris
2. How to read in comma-seperated values using pandas - https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html

In [11]:
dft_Iris = pd.read_csv("iris.data") # use the pandas pd.read_csv function to load in the iris.data file (where values are seperated by commas) and assign this to a variable for use later
print("The Iris data set is shown in the following data frame:")
display(dft_Iris) # display the dataframe to ensure it loaded in the correct format

The Iris data set is shown in the following data frame:


sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)  \
0                  5.1               3.5                1.4               0.2   
1                  4.9               3.0                1.4               0.2   
2                  4.7               3.2                1.3               0.2   
3                  4.6               3.1                1.5               0.2   
4                  5.0               3.6                1.4               0.2   
..                 ...               ...                ...               ...   
145                6.7               3.0                5.2               2.3   
146                6.3               2.5                5.0               1.9   
147                6.5               3.0                5.2               2.0   
148                6.2               3.4                5.4               2.3   
149                5.9               3.0                5.1               1.8   

            species  
0       Iris-setosa  
1       Iris-setosa  
2       Iris-setosa  
3       Iris-setosa  
4       Iris-setosa  
..              ...  
145  Iris-virginica  
146  Iris-virginica  
147  Iris-virginica  
148  Iris-virginica  
149  Iris-virginica  

[150 rows x 5 columns]

### Task 3.1 - Outputs a summary of each variable to a single text file

### Resources used for completion of this section:
1. 

### Task 3.2 - Saves a histogram of each variable to png files

### Resources used for completion of this section:
1. 

### Task 3.3 - Outputs a scatter plot of each pair of variables

### Resources used for completion of this section:
1. 

### Task 3.4 - Performs any other analysis you think is appropriate

### Resources used for completion of this section:
1. 

# The End